### PR TITLE
Add required permissions for simulated cluster & various fixes for metrics generation

### PR DESCRIPTION
**Pull request by jacobbaungard**
This PR contains a few various fixes that was needed to run the metric collector simulator and generate new metrics for it. See details in individual commits messages.

### CODE REVIEW

File: tools/simulator/metrics-collector/generate-metrics-data.sh [hunk 1]
```diff
@@ -40,47 +41,63 @@ if ! command -v kubectl &>/dev/null; then
     if [[ "$(uname)" == "Linux" ]]; then
       curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.21.0/bin/linux/amd64/kubectl
     elif [[ "$(uname)" == "Darwin" ]]; then
-      curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.21.0/bin/darwin/amd64/kubectl
```
Comment (philipgough) [inline tools/simulator/metrics-collector/generate-metrics-data.sh:43]: I previously created and consolidated the tooling into https://github.com/stolostron/multicluster-observability-operator/blob/main/scripts/install-binaries.sh

Would it help to extend that script with the requirements and call as needed?

Comment (jacobbaungard) [reply tools/simulator/metrics-collector/generate-metrics-data.sh:43]: Yes, that seems sensible, will take a look :+1:

Comment (jacobbaungard) [reply tools/simulator/metrics-collector/generate-metrics-data.sh:43]: Ok done!

**Comment by jacobbaungard** [comment]
/hold

The `$*` in the install-scrips causes some issues (probably just a warning) when calling these script with cmd line args. Trying to fix...

**Review by philipgough — APPROVED** [review]
/lgtm

**Pull request merged**